apps/admin-gui/src/app/shared/components/dialogs/add-role-dialog/add-role-dialog.component.html

```diff
@@ -1,47 +1,63 @@
 <div class="{{theme}}">
-  <h1 mat-dialog-title>{{'DIALOGS.ADD_ROLE.TITLE' | translate}}</h1>
+  <div mat-dialog-title>
+    <h1 mat-dialog-title>{{'DIALOGS.ADD_ROLE.TITLE' | translate}}</h1>
+    <perun-web-apps-immediate-filter
+      *ngIf="selectedRule?.primaryObject"
+      (filter)="filterValue = $event"
+      [placeholder]="'SHARED_LIB.PERUN.COMPONENTS.UNIVERSAL_OBJECTS_LIST.FILTER'"
+      class="font-size-1rem"></perun-web-apps-immediate-filter>
+  </div>
   <mat-spinner *ngIf="loading" class="mr-auto ml-auto"></mat-spinner>
   <div *ngIf="!loading" mat-dialog-content class="dialog-container">
-    <perun-web-apps-role-search-select
-      [roles]="rules"
-      (roleSelected)="selectedRule = $event; selectedEntity = null">
+    <mat-spinner *ngIf="loading" class="ml-auto mr-auto"></mat-spinner>
+    <perun-web-apps-role-search-select [roles]="rules" (roleSelected)="selectedRule = $event">
```

Comment:
Did you consider having this search select above the object filter? I feel like the filter should be the first thing above the list, also this way the search select wouldn't disappear when scrolling the list (see screenshot)
![Screenshot from 2022-12-23 13-04-56](https://user-images.githubusercontent.com/97669007/209333577-7611b22d-40a8-4745-b110-9f26dcfa6c68.png)